src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp

```diff
@@ -2391,6 +2394,30 @@ GUI_TEST_CLASS_DEFINITION(test_7548) {
     }
 }
 
+GUI_TEST_CLASS_DEFINITION(test_7550) {
+    auto clickFindProducts = [&os] {
+        if (GTUtilsMdi::activeWindow(os)->objectName() != "123 [cant_translate.fa]") {
```

Comment:
1. Do not close windows. If you need your window to be active: activate it like this
```GTUtilsMdi::activateWindow ```

+ I think you need to activate it only once, not for every notification stack close event? This will make this whole lambda a 1 liner: ```GTUtilsOptionPanelSequenceView::pressFindProducts(os);```

2. Remove lambda: this is a single-use 1-2-lines function. 
It breaks 'reading order' (I need to read here, next there, etc...) and does not solve either 'single-code-multi-use' or 'big-code-chunk-with-a-name' problems.

Reply:
> 1. Do not close windows. If you need your window to be active: activate it like this
>    `GTUtilsMdi::activateWindow `

Why?
> activate it only once, not for every notification stack close event

What does it mean? What is the notification stack close event? I close a window when a notification is clicked instead of the "Find product(s)" button.

>    It breaks 'reading order' (I need to read here, next there, etc...)

If the lambda were written just before the loop would it improve readability?

> does not solve either 'single-code-multi-use' or 'big-code-chunk-with-a-name' problems.

Why?

Reply:
Using `GTUtilsMdi::activateWindow` will slow down the test.

Reply:
Can't click 101 times when using `GTUtilsMdi::activateWindow`.

Reply:
Okay, I understand the problem. I don't know how it is on Linux and Mac, but on Windows with my screen resolution it goes like this:
1. The "Find product(s)" button is clicked 10 times.
2. On the 11th, not the button is pressed, but a notification.
3. The "In Silico PCR" task report window opens.
4. Notifications disappeared because a new window opened.
5. The `GTUtilsMdi::activateWindow` method is called. It tries to open a window via MainMenuBar. The mouse clicked on "&Window".
6. Notifications appear because 1 notification is outdated and the number of notifications displayed on the screen has changed: they need to be redrawn.
7. Appeared notifications interrupt the called submenu.
8. The mouse keeps moving as if the submenu is open, but nothing really happens: the windows don't switch.

I can try switching between tabbed windows using Ctrl+Tab. Or close them.

Reply:
Lambda removed.

Reply:
> Why?
The test should have very precise logic: not "close whatever is opened", but "close window X", "activate window Y". This way the test will be more sensitive to changes.